src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let cloneShadow = {};
+  const clone = {};
+  const result = [];
+
+  Object.assign(clone, state);
+
+  // Main Loop:
+  for (const action of actions) {
+    cloneShadow = {};
+
+    switch (action['type']) {
+      // For the type of 'addProperties':
+      case 'addProperties':
+        for (const extraItem in action['extraData']) {
+          clone[extraItem] = action['extraData'][extraItem];
+        }
```

Comment:
You can use Object.assign for this case, instead of loop

Reply:
agree, let's use Object.assign for adding new values to the object